### PR TITLE
Refactor KIR Step 4 - Remove kir::Bool,Double,Int,NamedScalar.

### PR DESCRIPTION
**Pull request by csarofeen**
This one was relatively straight forward, the only caveat is I believe:
`NamedScalar::getParallelIndex` and `NamedScalar::getParallelDim` are not supported in Kernel at the moment, but I didn't see any uses there anyways. I mild protection gets added at some point but can't remember which PR its added in:
https://github.com/csarofeen/pytorch/pull/1344/files#diff-296b2445f5a818c3240b26185b37b309fa554f56b909f73109917cbdfd580333R1660

### CODE REVIEW

**Comment by csarofeen** [comment]
Sorry Masaki, didn't mean to add you as a reviewer.

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -1335,7 +1335,7 @@ TEST_F(NVFuserTest, FusionForLoop_CUDA) {
   BinaryOp* op = static_cast<BinaryOp*>(TV2->definition();
   fusion.addOutput(TV2);
 
-  auto fl = new kir::ForLoop(new kir::Int(c10::nullopt), ID0, {op});
+  auto fl = new kir::ForLoop(new Int(c10::nullopt), ID0, {op});
```
Comment (naoyam) [inline test/cpp/jit/test_gpu.cpp]: I was puzzled why `kir::ForLoop` could be constructed this way without using the passkey, and just realized this is commented out.  Do we want to keep this test? Or is it considered stale?

**Review by naoyam — APPROVED** [review]
LGTM. Just one minor comment.

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (csarofeen) [reply test/cpp/jit/test_gpu.cpp]: Test doesn't make much sense to me, removing.

**Pull request merged**